generated/nidaqmx/_task_modules/triggering/start_trigger.py

```diff
@@ -987,6 +987,43 @@ def cfg_anlg_edge_start_trig(
         self._interpreter.cfg_anlg_edge_start_trig(
             self._handle, trigger_source, trigger_slope.value, trigger_level)
 
+    def cfg_anlg_multi_edge_start_trig(
+            self, trigger_sources, trigger_slope_array,
+            trigger_level_array=None):
+        """
+        Configures the task to start acquiring or generating samples
+        when any of the configured analog signals cross the respective
+        levels you specified. Multi-edge triggering treats the
+        configured triggers as if a logical OR is applied.
+
+        Args:
+            trigger_sources (str): Is the name of a virtual channel or
+                terminal where there is an analog signal to use as the
+                source of the trigger.
+            trigger_slope_array (List[nidaqmx.constants.Slope]): 
+                Specifies on which slope of the signal to start
+                acquiring or generating samples when the signal crosses
+                **level**.
+            trigger_level_array (Optional[List[float]]): Specifies at
+                what threshold to start acquiring or generating samples.
+                Specify this value in the units of the measurement or
+                generation. Use **slope** to specify on which slope to
+                trigger at this threshold.
+        """
+        if trigger_slope_array is None:
+            trigger_slope_array = []
+
+        if trigger_level_array is None:
+            trigger_level_array = []
+
+        trigger_slope_array = numpy.array([p.value for p in trigger_slope_array], dtype=numpy.int32)
+        trigger_level_array = numpy.array(trigger_level_array, dtype=numpy.float64)
+
+
+        self._interpreter.cfg_anlg_multi_edge_start_trig(
+            self._handle, trigger_sources, trigger_slope_array,
```

Comment:
@bkeryan @zhindes 
I need advice on the `trigger_sources`. In the test [test___start_trigger___cfg_anlg_multi_edge_start_trig___no_errors](https://github.com/ni/nidaqmx-python/pull/535/files#diff-05b936155dab8db8b3bc0dfce153a40c0021599197bbba8487a4b0ec5e4d6872R154), giving a list of str for  trigger_sources will get type error. By referring [set_analog_power_up_states_with_output_type](https://github.com/ni/nidaqmx-python/blob/0f65354421c5de8799eef8e1343db6f16af95524/generated/nidaqmx/system/system.py#L405), I have to provide `flatten_trigger_sources`.

Should we set `'python_codegen_method': 'custom'` to `CfgAnlgMultiEdgeStartTrig` metadata and then define the function in mako file like `set_analog_power_up_states_with_output_type`? So that we can process `trigger_sources` before passing it to interpreter.

Reply:
An alternative implementation for all of this would be to create a cluster of `{ trigger_source: string, trigger_slope: enum, trigger_level: float}` and accept an array of those, just like the power up states. However, this is a very niche API only supported by one device. We don't need to go above and beyond for this. Forcing the user to call `utils.flatten_channel_string` is fine by me. If, in the future, we really cared, we could add a new Python API that does that, but I guarantee you we won't do that :)

Reply:
Note: @bkeryan just noticed that we have precedence for exactly what you have implemented in .NET.
* [StartTrigger.ConfigureAnalogMultiEdgeTrigger](https://www.ni.com/docs/en-US/bundle/ni-daqmx-net-framework-45-api-ref/page/ninetdaqmxfx45ref/html/m_nationalinstruments_daqmx_starttrigger_configureanalogmultiedgetrigger.htm)
* [ReferenceTrigger.ConfigureAnalogMultiEdgeTrigger](https://www.ni.com/docs/en-US/bundle/ni-daqmx-net-framework-45-api-ref/page/ninetdaqmxfx45ref/html/m_nationalinstruments_daqmx_referencetrigger_configureanalogmultiedgetrigger.htm)

Those methods require the user to flatten the sources.